partitioner/parsers/open_street_map.py

```diff
@@ -28,3 +28,44 @@ def get_node_info(self, line: Text) -> Tuple[int, float, float]:
 
     def get_edge_info(self, line: Text) -> Tuple[int, int, float]:
         pass
+
+    def is_in_partition(lon, lat, partition):
+        x_min, x_max = partition[0]
+        y_min, y_max = partition[1]
+        return x_min < lon <= x_max and y_min < lat <= y_max
+
+    def parse_way(way_lines):
+        tag_regex = re.compile(r'\s*<tag k="(\w+)" v="(.+)"\s*\/>')
+        node_regex = re.compile(r'\s*<nd ref="(\d+)"\s*\/>')
+        way_begin = re.compile(r'\s*<way [^>]*>\s*')
+        way_end = re.compile(r'\s*<\/way>\s*')
```

Comment:
też te regexy można by kompilować w `_init_`, idk czy to ma znaczący performance impact